Add an optional prop to DocumentHead to set the raw window title

DanReyLop opened the pull request:
#### Changes proposed in this Pull Request

The `<DocumentHead>` component has a `title` prop, to set the tab title in the browser (`document.title`). However, on top of whatever title is passed, it applies some extra formatting. For example, passing `Hello World` as a title can result in a window title being `Hello World - WordPress.com (3)`.

This PR adds a new optional prop, `formattedTitle`. If present, that string will be used as the raw window title, without any additions.

It also adds an optional `pageTitle` prop to the `JetpackConnectMainWrapper` component, that makes use of that prop.

#### How to test

There's no code that calls `JetpackConnectMainWrapper` with a `pageTitle` prop, or `DocumentHead` with a `formattedTitle` prop, so there's no way to test this.

Note: This is a required refactor for the new "Woo DNA" flow, implemented in #41798

DanReyLop committed:
Move the styles for the login social divider to the login block's styles

DanReyLop committed:
Make the LoginBlock accept an optional "userEmail" prop to prefill the field

DanReyLop committed:
Make the SignupBlock accept an optional "handleLogin" function prop to intercept the "is this you? log in" link

DanReyLop committed:
Make the SignupBlock accept an optional "footerLink" prop to override the default footer links.

DanReyLop committed:
Extracted the "Back to WP-Admin" link rendering to a separate function

DanReyLop committed:
Remove unused prop

DanReyLop committed:
Replace the "isWoo" parameter of the login URL builder with a more generic "from" parameter

DanReyLop committed:
Refactor: Add an optional prop to DocumentHead to set the window title without any extra formatting

Aurorum commented on client/components/data/document-head/index.jsx:
```diff
@@ -81,6 +81,7 @@ class DocumentHead extends Component {
 
 DocumentHead.propTypes = {
 	title: TranslatableString,
+	formattedTitle: TranslatableString,
```
I think it may be worth adding this to the existing list of props within the documentation: https://github.com/Automattic/wp-calypso/blob/master/client/components/data/document-head/README.md

DanReyLop replied:
Good catch, I missed that! Done in https://github.com/Automattic/wp-calypso/pull/41803/commits/5751f65a74b718e52a22619b646bd2dedc91fdb9

DanReyLop committed:
Updated the documentation

DanReyLop committed:
Fixed code style

DanReyLop committed:
Fixed code style

vbelolapotkov reviewed (commented):
Thank you @DanReyLop for that refactoring! Changes look good and fine to be merged like this.
However, to be honest, having two competing title props slightly bothers me. What do you think on having a single `title` prop and a flag, e.g. `skipTitleFormatting` or `titleIsFormatted`. This way IMO it's more explicit that `title` is always  a title and flag emphasizes desired behavior. What do you think?

DanReyLop committed:
Add a "skipTitleFormatting" flag as a prop, instead or using 2 competing "title" props

DanReyLop commented:
@vbelolapotkov I implemented your suggestion in https://github.com/Automattic/wp-calypso/pull/41803/commits/383c41ae699245cc7b480ed44592c72a630262a2

I also rebased & force-pushed all the branches that are on top of this one, so `woodna/woodna-flow` has **all** the changes included.

jsnajdr commented on client/jetpack-connect/main-wrapper.jsx:
```diff
@@ -46,7 +47,10 @@ export class JetpackConnectMainWrapper extends PureComponent {
-				<DocumentHead title={ translate( 'Jetpack Connect' ) } />
+				<DocumentHead
+					title={ pageTitle || translate( 'Jetpack Connect' ) }
+					skipTitleFormatting={ Boolean( pageTitle ) }
```
Just to verify: the fallback title, when `pageTitle` prop is not supplied, is supposed to be formatted? E.g., `Jetpack Connect --- WordPress.com (6)`?

DanReyLop replied:
Yes, that's the existing behaviour.

vbelolapotkov reviewed (approved):
LGTM! Thank you @DanReyLop !

jsnajdr reviewed (approved):
Nice! I also like the `skipTitleFormatting` API better than the original "raw title" one.
